src/mechanics/timeIntegration/NewmarkDirect.cpp

```diff
@@ -1,499 +1,182 @@
-            {
-
-                // ******************************************************
-                mStructure->Evaluate(inputMap, evaluate_Hessian0_Hessian1);
-                // ******************************************************
-
-                if (mCheckCoefficientMatrix)
-                    mStructure->ElementCheckHessian0(1.e-6, 1.e-8);
-
-
-                // ******************************************************
-                delta_dof_dt0.J = BuildHessianModAndSolveSystem(hessian0, hessian1, hessian2, residual_mod, timeStep);
-                delta_dof_dt0.K = cmat * delta_dof_dt0.J * (-1.);
-                ++mIterationCount;
-                // ******************************************************
-
-
-                // perform a line search
-                double alpha = 1;
-                BlockScalar trialNormResidual(mStructure->GetDofStatus());
-                do
-                {
-                    // calculate line search trial state
-                    trial_dof_dt0 = dof_dt0 + delta_dof_dt0 * alpha;
-                    if (mStructure->GetNumTimeDerivatives() >= 1)
-                        trial_dof_dt1 = dof_dt1 + delta_dof_dt0 * (alpha * mGamma / (timeStep * mBeta));
-                    if (mStructure->GetNumTimeDerivatives() >= 2)
-                        trial_dof_dt2 = dof_dt2 + delta_dof_dt0 * (alpha / (timeStep * timeStep * mBeta));
-
-                    MergeDofValues(trial_dof_dt0, trial_dof_dt1, trial_dof_dt2, false);
-
-                    // ******************************************************
-                    mStructure->Evaluate(inputMap, evaluate_InternalGradient);
-                    // ******************************************************
-
-                    residual = CalculateResidual(intForce, extForce, hessian2, trial_dof_dt1, trial_dof_dt2);
-                    residual.ApplyCMatrix(residual_mod, cmat);
+                trial_dof_dt2 =
+                        dof_dt[2] +
+                        delta_dof_dt0 * (alpha / (mTimeControl.GetTimeStep() * mTimeControl.GetTimeStep() * mBeta));
 
-                    trialNormResidual = residual_mod.CalculateInfNorm();
+            MergeDofValues(trial_dof_dt0, trial_dof_dt1, trial_dof_dt2, false);
 
-                    mStructure->GetLogger() << "[Linesearch a=" << std::to_string(alpha).substr(0, 6)
-                                            << "] Trial residual: " << trialNormResidual << "\n";
+            const auto intForce = EvaluateInternalGradient();
 
-                    alpha *= 0.5;
+            structureResidual = CalculateResidual(intForce, extForce, hessians[2], trial_dof_dt1, trial_dof_dt2);
+            residual = Assembler::ApplyCMatrix(structureResidual, constraintMatrix);
 
-                } while (mPerformLineSearch && alpha > mMinLineSearchStep &&
-                         trialNormResidual > (1. - alpha) * normResidual);
+            trialNormResidual = residual.CalculateInfNorm();
```

Comment:
Is the InfNorm the standard to check convergence? IMO this would mean that by reducing the mesh size the acceptable error in the solution increases.